File: 1310-watering-plants/watering-plants.py
class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        steps = 1
        curcapacity = capacity
        
        for i in range(len(plants)-1):
            curcapacity -= plants[i]
            if plants[i+1] > curcapacity:
                curcapacity = capacity
                steps += i*2+2
            steps+=1

        return steps

File: 1310-watering-plants/watering-plants.cpp
class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity) 
    {
        int steps = 1, curcapacity = capacity;
        
        for (int i = 0; i < plants.size()-1; i++)
        {
            curcapacity -= plants[i];
            if (plants[i+1] > curcapacity)
            {
                curcapacity = capacity;
                steps += (i+1)*2;
            }
            steps++;
        }
        return steps;
    }
};